hazelcast-sql-core/src/test/java/com/hazelcast/sql/impl/calcite/opt/physical/index/PhysicalSortIndexTest.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.calcite.opt.physical.index;
+
+import com.hazelcast.config.IndexType;
+import com.hazelcast.sql.impl.calcite.opt.physical.MapIndexScanPhysicalRel;
+import com.hazelcast.sql.impl.calcite.opt.physical.MapScanPhysicalRel;
+import com.hazelcast.sql.impl.calcite.opt.physical.ProjectPhysicalRel;
+import com.hazelcast.sql.impl.calcite.opt.physical.RootPhysicalRel;
+import com.hazelcast.sql.impl.calcite.opt.physical.SortPhysicalRel;
+import com.hazelcast.sql.impl.calcite.opt.physical.exchange.SortMergeExchangePhysicalRel;
+import com.hazelcast.sql.impl.calcite.schema.HazelcastSchema;
+import com.hazelcast.sql.impl.calcite.schema.HazelcastTable;
+import com.hazelcast.sql.impl.schema.map.MapTableIndex;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.annotation.ParallelJVMTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.apache.calcite.schema.Table;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.hazelcast.sql.impl.type.QueryDataType.INT;
+import static java.util.Arrays.asList;
+import static java.util.Collections.singletonList;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelJVMTest.class})
+public class PhysicalSortIndexTest extends IndexOptimizerTestSupport {
```

Comment:
We need more tests to handle typical cases here:
1. Tests when there is no merge phase
2. Tests when sorting is combined with index lookups 
3. Negative tests, e.g. for HASH indexes, missing indexes, lost collations

Reply:
Added in https://github.com/hazelcast/hazelcast/pull/18013/commits/fa9da4a373283a555a65e2e37f19db92608dbf3b